model/comments.model.js

```diff
@@ -0,0 +1,25 @@
+const db = require("../db/connection.js");
+const { fetchArticlesById } = require("./articles.model.js");
+
+exports.fetchComments = (article_id) => {
+  return db
+    .query("SELECT comments.* FROM comments WHERE comments.article_id=$1", [
+      article_id,
+    ])
+    .then((results) => {
+      
+      if (results.rows.length === 0) {
+        return fetchArticlesById(article_id);
+      }else{
+      return results.rows;
+      }
+    })
+    .then((results) => { 
+        if(Array.isArray(results)){
+
+      return results;
+        } else{
+            return []
+        }
```

Comment:
1. return fetchArticlesById(article_id) will check if article exists. If it is rejected, it won't move to the .then block on 17, will throw an error instead. If it's fulfilled, it will move to the .then block, results on 17 should be an object. So the if statement on 18 won't be executed anyway. Can remove it.
2. I would say invoking a model inside another model is not the best practice. Would be better if you invoke fetchArticlesById inside your controller